e2e/machine_migration_test.go

```diff
@@ -111,6 +113,275 @@ var _ = Describe("[sig-cluster-lifecycle][OCPFeatureGate:MachineAPIMigration] Ma
 			})
 		})
 	})
+
+	var _ = Describe("Update MAPI Machine", Ordered, func() {
+		Context("with spec.authoritativeAPI modification", func() {
+			It("should allow modification of authoritativeAPI from ClusterAPI to MachineAPI", func() {
+				By("Attempting to modify authoritativeAPI from ClusterAPI to MachineAPI")
+
+				// Create a new machine for this test to avoid affecting other tests
+				testMachineName := "machine-auth-change-83955"
+				testMachine := createMAPIMachineWithAuthority(ctx, cl, testMachineName, machinev1beta1.MachineAuthorityClusterAPI)
+
+				// Wait for the CAPI machine to be created (any status is fine for this test)
+				Eventually(func() error {
+					_, err := capiframework.GetMachine(cl, testMachineName, capiframework.CAPINamespace)
+					return err
+				}, capiframework.WaitLong, capiframework.RetryLong).Should(Succeed(), "CAPI Machine should be created")
+
+				// Add a small delay to ensure any pending updates are processed
```

Comment:
See above, all of this could be a komega Update, which wraps an additional `Get` and can be called from directly within an Eventually()